outlines/blaze.md

```diff
@@ -0,0 +1,56 @@
+# Blaze
+
+1. Introduction to Blaze -- the tracker-backed handlebars like templating syntax
+  1. Example of spacebars syntax, data context + helpers
+  2. Data contexts and access - name lookup (`.`, `this`, `..`, `[0]`), and null values
+  3. Helpers, arguments, options
+  4. Inclusion of other templates + arguments, passing data contexts
+  5. Helpers inside tags -- returning strings and objects, `checked={{isChecked}}`
+  6. Nested helpers + sub expressions
+  7. Block helpers (you can create them with templates, see 10. below)
+  8. Safestrings and `{{{`
+  9. Builtin block helpers
+    1. `{{#if/unless}}`
+    2. `{{#with}}`
```

Comment:
> In some way, logic should be in template helpers in my opinion. Putting too much of it into templates, just so that you can then pass a value to a template helper so that it does not have to access the data context is probably bad.

I don't see what you mean here by "logic" - I wouldn't say managing which data is available where entails logic, since it's an inherent property of the view tree.

> Maybe a method on template instances like lookup could be helpful

Yes, I believe this is the best solution as well. I'd love to have a sane version of `Template.currentData` with exactly the properties you describe. If we can agree on the design, I'll 100% accept a PR for that.

There are some design questions though, and we should talk about them in a different thread of you would like to start one.